lib/AutoPodArgsPass.cpp

```diff
@@ -52,6 +52,39 @@ PreservedAnalyses clspv::AutoPodArgsPass::run(Module &M,
   return PA;
 }
 
+namespace {
+bool FunctionContainsImageChannelGetter(Function *F) {
+  std::set<Function *> visited_fct;
+  SmallVector<Function *, 1> fcts_to_visit;
+  SmallVector<Function *, 1> next_fcts_to_visit;
+  fcts_to_visit.push_back(F);
+  while (!fcts_to_visit.empty()) {
+    for (auto *fct : fcts_to_visit) {
+      visited_fct.insert(fct);
+      for (auto &BB : *fct) {
+        for (auto &I : BB) {
+          if (auto call = dyn_cast<CallInst>(&I)) {
+            auto Name = call->getCalledFunction()->getName();
+            if (Name.contains("get_image_channel_order") ||
+                Name.contains("get_image_channel_data_type")) {
+              return true;
+            } else {
+              Function *f = call->getCalledFunction();
+              if (visited_fct.count(f) == 0) {
+                next_fcts_to_visit.push_back(f);
+              }
+            }
+          }
+        }
+      }
+    }
+    fcts_to_visit = std::move(next_fcts_to_visit);
+    next_fcts_to_visit.clear();
```

Comment:
This is triggering the use-after-move Clang check.

https://clang.llvm.org/extra/clang-tidy/checks/bugprone/use-after-move.html 

Seems the declaration of next_fcts_to_visit should be moved inside the loop.  I don't think this is performance critical to warrant tolerating the bug.